### PR TITLE
[Azure Search] Introduce initial set of changes for the preview SDK

### PR DESCRIPTION
**Pull request by arv100kri**
1. Remove a bunch of obsolete types 
    - `DataSourceType` and `NamedEntityRecognitionSkill` removed via "swagger"
    - Other index related obsolete types directly deleted in the SDK
2. Replace a bunch of customized enums via extensible enums, generated directly from swagger.
    - `TokenFilterName`, `CharFilterName`, `TokenizerName`, `DataSourceType` and
    - Language types for skills
3. Minor doc comment fixes also pulled from swagger
4. Re-ran all existing tests (removed 1 unnecessary test) + update session records

Upstream swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/7751

### CODE REVIEW

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexes/Models/StandardTokenizer.cs [hunk 1]
```diff
@@ -1,60 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexes/Models/StandardTokenizer.cs]: Let's hold off on deleting the "obsolete" analysis components for now. We should first check that they're no longer part of any index, but I'm not sure we have the telemetry yet to answer this question. It isn't a priority right now, so for now just keep them around.

File: sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/TokenFilterName.cs [hunk 2]
```diff
@@ -29,181 +29,214 @@ private TokenFilterName(string underlyingValue)
...
+        /// See
+        /// http://lucene.apache.org/core/4_10_3/analyzers-common/org/apache/lucene/analysis/in/IndicNormalizationFilter.html
         /// </summary>
         public static readonly TokenFilterName IndicNormalization = "indic_normalization";
 
         /// <summary>
         /// KeywordRepeat
         /// </summary>
-        public static readonly TokenFilterName EmitseachincomingtokentwiceonceaskeywordandonceasnonKeyword = "keyword_repeat";
+        public static readonly TokenFilterName EmitseachincomingtokentwiceonceaskeywordandonceasnonKeywordSeehttpLuceneapacheorgCore4103AnalyzersCommonOrgApacheLuceneAnalysisMiscellaneousKeywordRepeatFilterhtml = "keyword_repeat";
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/TokenFilterName.cs]: Oops?

**Review by bitsforthought — CHANGES_REQUESTED** [review]
One of the token filter names looks very wrong (probably a Swagger bug).

**Comment by bacongobbler** [comment]
Not sure why I was requested to review this. Mind removing me? Thanks!

File: sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/TokenFilterName.cs (hunk 2 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/TokenFilterName.cs]: Yup, I see why... will fix the swagger!

File: sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexes/Models/StandardTokenizer.cs (hunk 1 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Service/src/Customizations/Indexes/Models/StandardTokenizer.cs]: Got it, I'll remove other methods/parameters...

**Comment by arv100kri** [comment]
I didn't add you, must be something with the automated assignee system. I am not sure how to remove you though...

---
In reply to: [552695760](https://github.com/Azure/azure-sdk-for-net/pull/8733#issuecomment-552695760) [](ancestors = 552695760)

File: sdk/search/Microsoft.Azure.Search.Data/src/Customizations/SearchIndexClient.Customization.cs [hunk 3]
```diff
@@ -70,6 +70,18 @@ public SearchIndexClient(string searchServiceName, string indexName, SearchCrede
         /// </summary>
         public bool UseHttpGetForQueries { get; set; }
 
+        /// <summary>
```
Comment (brjohnstmsft) [inline sdk/search/Microsoft.Azure.Search.Data/src/Customizations/SearchIndexClient.Customization.cs]: I don't mind if this is deleted.

**Review by bitsforthought — APPROVED** [review]
`TargetDifferentIndex` can be safely deleted. Otherwise, LGTM

File: sdk/search/Microsoft.Azure.Search.Data/src/Customizations/SearchIndexClient.Customization.cs (hunk 3 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Data/src/Customizations/SearchIndexClient.Customization.cs]: I did in the latest commit :)

**Comment by arv100kri** [comment]
> `TargetDifferentIndex` can be safely deleted. Otherwise, LGTM

Obviously I am waiting to regenerate from the master branch before I make this ready for checkin...

**Pull request merged**